### ISSUE TITLE
Ignore shellcheck when using PowerShell as SHELL

### ISSUE THREAD

**Issue by 4x0v7** [hadolint/hadolint#246]
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
When using `SHELL [ "pwsh", "-c" ]`, the commands in a `RUN` should not be run through ShellCheck

### Actual behavior
```
/dev/stdin:4 SC1008 This shebang was unrecognized. Note that ShellCheck only handles sh/bash/dash/ksh.
/dev/stdin:4 DL4006 Set the SHELL option -o pipefail before RUN with a pipe in it
```

### Steps to reproduce the behavior

Consider the following Dockerfile:

```
FROM mcr.microsoft.com/powershell:ubuntu-16.04
SHELL [ "pwsh", "-c" ]
RUN Get-Variable PSVersionTable | Select-Object -ExpandProperty Value
```

This Dockerfile builds ok. Information on the `pwsh` Linux binary is available here:
https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_pwsh

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.10.4-3-g20c5c3b
```

**Dockerfile** (if relevant)

```Dockerfile
FROM mcr.microsoft.com/powershell:ubuntu-16.04
SHELL [ "pwsh", "-c" ]
RUN Get-Variable PSVersionTable | Select-Object -ExpandProperty Value
```

**Comment by lorenzo** [comment]
Thanks for reporting, I will look into this

**Comment by blueharvest-ben** [comment]
Hi this is useful, thanks! but many windows dockerfile images use "powershell.exe" rather than "pwsh" as the shell. Can you possibly extend the code to include this for the same reasons?

**Comment by okainov** [comment]
I believe, the same should be applied for cmd...

```
SHELL ["cmd.exe", "/s", "/c"]
```
